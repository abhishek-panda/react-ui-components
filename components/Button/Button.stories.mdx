import { Meta, Story, Preview, Props} from '@storybook/addon-docs/blocks';
import Button from './index';

<Meta title="Button" component={Button} />

# Checkbox

With `MDX` we can define a story for `Checkbox` right in the middle of our
markdown documentation.

<Preview>
  <Story name="standard">
    <Button>
      <span role="img" aria-label="so cool">
        😀 😎 👍 💯
      </span>
    </Button>
  </Story>
</Preview>

<Props of={Button} />