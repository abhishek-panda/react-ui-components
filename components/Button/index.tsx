import * as React from 'react';
import bind from '../utils/bind';

interface Props {
    children : React.ReactChild,
    msg ?: string
}


interface State {
    count : number
}

export class ButtonComponent extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);
        this.state = {
            count: 0,
        };
    }

    @bind
    clickHandler () {
        console.log(this.state);
    }

    render() {
        return <button onClick={this.clickHandler}>{this.props.children}</button>
    }
}


export default ButtonComponent;