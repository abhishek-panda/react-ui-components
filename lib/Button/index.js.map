{"version":3,"file":"index.js","sourceRoot":"","sources":["../../components/Button/index.tsx"],"names":[],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;AAAA,2CAA+B;AAC/B,oDAA8B;AAW9B;IAAqC,mCAA6B;IAChE,yBAAY,KAAY;QAAxB,YACE,kBAAM,KAAK,CAAC,SAIb;QAHC,KAAI,CAAC,KAAK,GAAG;YACX,KAAK,EAAE,CAAC;SACT,CAAC;;IACJ,CAAC;IAGO,sCAAY,GAApB;QACU,IAAA,KAAK,GAAK,IAAI,CAAC,KAAK,MAAf,CAAgB;QAC7B,OAAO,KAAK,CAAC;IACf,CAAC;IAED,gCAAM,GAAN;QACU,IAAA,QAAQ,GAAK,IAAI,CAAC,KAAK,SAAf,CAAgB;QAChC,OAAO,gCAAQ,IAAI,EAAC,QAAQ,EAAC,OAAO,EAAE,IAAI,CAAC,YAAY,IAAG,QAAQ,CAAU,CAAC;IAC/E,CAAC;IARD;QADC,cAAI;;;;uDAIJ;IAMH,sBAAC;CAAA,AAlBD,CAAqC,KAAK,CAAC,SAAS,GAkBnD;AAlBY,0CAAe;AAoB5B,kBAAe,eAAe,CAAC"}