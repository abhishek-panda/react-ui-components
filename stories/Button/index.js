import React from 'react';
import { action } from '@storybook/addon-actions';
import { Button } from '@storybook/react/demo';

export const myBtn = () => (
  <Button onClick={action('clicked')}>
    <span role="img" aria-label="so cool">
      😀 😎 👍 💯
    </span>
  </Button>
);

export default {
  title: 'Button',
};