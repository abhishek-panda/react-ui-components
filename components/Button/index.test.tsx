import React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from './index';

describe('<Button />', () => {
  describe('Taking Snaphot', () => {
    let wrapper: ReactWrapper;
    const component = <Button>Hello</Button>;
    beforeEach(() => {
      wrapper = mount(component);
    });

    it('should match its snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
