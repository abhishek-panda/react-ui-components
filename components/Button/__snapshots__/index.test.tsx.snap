// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Taking Snaphot should match its snapshot 1`] = `
<ButtonComponent>
  <button
    onClick={[Function]}
    type="button"
  >
    Hello
  </button>
</ButtonComponent>
`;
